lxd/certificates.go

```diff
@@ -1180,11 +1180,11 @@ func certificateDelete(d *Daemon, r *http.Request) response.Response {
 
 func certificateValidate(networkCert *shared.CertInfo, cert *x509.Certificate) error {
 	if time.Now().Before(cert.NotBefore) {
-		return fmt.Errorf("The provided certificate isn't valid yet")
+		return api.NewStatusError(http.StatusBadRequest, "The provided certificate isn't valid yet")
 	}
 
 	if time.Now().After(cert.NotAfter) {
-		return fmt.Errorf("The provided certificate is expired")
+		return api.NewStatusError(http.StatusBadRequest, "The provided certificate is expired")
```

Comment:
Hmm yes to the second one (RSA key too weak) but the second one is a type assertion that we don't expect to fail since we've already checked the public key algorithm of the cert, so I'll leave that as a plain error that will become a 500.